### DISCUSSION TITLE
Using Delphi controls: DBEdit, DBText, etc?

### DISCUSSION THREAD

**Discussion by KBracknell** [SchneiderInfosystems/FB4D#184] [General]
Hi Christoph, is it possible to associate Firebase/Firestore database table fields with Delphi's database controls such as DBEdit, DBText, etc.
Thanks,
Kim.

**Comment by SchneiderInfosystems** [comment]
Hi Kim
No both DB are NSQL DBs and does not support TDataser because it stores JSON Objects not flat records For Firestore the best approach is using the new object to document wrapper. Check the wiki and the sample app Samples/FSObject2Doc or Samples/FSObj2Doc_VCL.
Br Christoph
